recipes/sdl_image/all/conanfile.py

```diff
@@ -50,9 +54,13 @@ class SDLImageConan(ConanFile):
         "xpm": True,
         "xv": True,
         "with_libjpeg": True,
+        "with_libjpeg_dynamic_load": False,
```

Comment:
When `with_libjpeg_dynamic_load` is true, you need to make sure in `package_info`, that the `SDL2_image` target does not explicitly depend/link to libjpeg.
The library is then dynamically loaded using `dlopen` + `dladdr`.

You might need to introduce a dummy target to collect all "non-used" dynamically loaded dependencies (unless conan 2 has a feature for this)

Reply:
> You might need to introduce a dummy target to collect all "non-used" dynamically loaded dependencies (unless conan 2 has a feature for this)

I think there is something in conan v2: https://github.com/conan-io/conan/issues/12435

Reply:
I thought setting `headers`/`libs` of `self.requires` only influences what properties are set on CMake targets (`INTERFACE_LINK_LIBRARIES` for `libs`, and `INTERFACE_INCLUDE_DIRECTORIES` for `headers`).
It's okay to have a target `JPEG::JPEG` with or without libraries, but we don't want to link against it.

Reply:
I should also remove link to dependencies in custom CMakeLists of this recipe when dynamic load is enabled.

Reply:
I think you're correct but I think they should not be used here. (thanks for describing them)

In the upstream cmake script, it needs a functional `FindXXX.cmake` module (or `XXXConfig.cmake`) because it uses the library visible through `JPEG::JPEG` as name for what jpeg library to load with `dlopen`.
(For Windows, it uses the import library to know what dll whould be loaded using `LoadLibraryA`).

Users of `SDL2_image` can also not use the properties you just described (even not `transitive_libs`), because SDL2_image can be configured with a mixture of dynamically loaded and hard linked dependencies. It depends on what options were used to build SDL2_image.

So I think something as follows is required in `package_info`:
```python
if self.options.with_libjpeg_dynamic_load:
    self.package_info.modules["_sdl_image"].dependencies = "libjpeg"
else:
    self.package_info.modules["sdl_image"].dependencies = "libjpeg"
```